subsys/bluetooth/mesh/rpl.c

```diff
@@ -160,12 +155,23 @@ bool bt_mesh_rpl_check(struct bt_mesh_net_rx *rx,
 	return true;
 }
 
+static void rpl_clear_pending_handler(struct k_work *work)
+{
+	bt_mesh_settings_store_cancel(BT_MESH_SETTINGS_RPL_PENDING);
```

Comment:
@PavelVPV  
> @LingaoM will bt_mesh_settings_store_pending call after bt_mesh_rpl_clear in net.c solve the problem? I've introduced it here https://github.com/zephyrproject-rtos/zephyr/pull/47608/files to solve the same problem for bt_mesh_reset. The other solution could be to call bt_mesh_rpl_reset twice, but this doesn't look neat.

Here

Reply:
From what I see, settings will be scheduled immediately due to iv index and seq number store here: https://github.com/zephyrproject-rtos/zephyr/blob/88aa873fbc544b99c1e48e37d5c74662f90b3de0/subsys/bluetooth/mesh/net.c#L340 So I don't see point in additional k_work item and the only problem is that `BT_MESH_VALID` flag doesn't let to clear rpl entries in `bt_mesh_rpl_pending_store`.

Reply:
Just store immediately

Reply:
However, will this storage cause the task stack occupation of BT RX to increase and the block time to increase to operate flash, but this usage seems to have been used in other places, so do it?

Reply:
> will this storage cause the task stack occupation of BT RX to increase 

Yes.

> the block time to increase to operate flash

Shouldn't be matter since this is cooperative thread as well as system workqueue.

It looks like an additional flag (probably along with `BT_MESH_VALID`), should be enough to set in `bt_mesh_rpl_clear` and check in `bt_mesh_rpl_pending_store`.